local-macros/src/lib.rs

```diff
@@ -74,15 +74,100 @@ pub fn ast_type(_attr: TokenStream, item: TokenStream) -> TokenStream {
                 }
             }
         }
-        _ => panic!("Expected struct"),
+        Enum(DataEnum { variants, .. }) => {
+            let variant_names = variants.iter().map(|variant| &variant.ident);
+            let variant_names_2 = variant_names.clone();
+            let variant_names_3 = variant_names.clone();
+            let variant_names_4 = variant_names.clone();
+            let variant_names_5 = variant_names.clone();
+            let variant_names_6 = variant_names.clone();
+            let variant_names_7 = variant_names.clone();
+            let variant_names_8 = variant_names.clone();
+            let variant_names_9 = variant_names.clone();
+            let variant_names_10 = variant_names.clone();
+            let variant_names_11 = variant_names.clone();
+
+            quote! {
+                impl crate::NodeInterface for #ast_type_name {
+                    fn kind(&self) -> crate::SyntaxKind {
+                        match self {
+                            #(#ast_type_name::#variant_names(nested) => nested.kind()),*
```

Comment:
This is the `quote` syntax for looping through an iterable

Reply:
Is there any way to iterate over references (to avoid all the copies of `variant_names`?).

Reply:
Ya I'm not sure, didn't see anything to that effect in glancing at the `quote` docs